hpcrawler.py

```diff
@@ -0,0 +1,364 @@
+        if not os.path.exists(self._path):
+            os.makedirs(self._path)
+
+        profile.set_preference('browser.download.dir', os.path.abspath(self._path))
+        profile.set_preference('browser.helperApps.neverAsk.saveToDisk', 'application/octet-stream')
+        driver = webdriver.Firefox(firefox_profile=profile)
+        driver.set_window_size(50, 1000)
+        self._driver_status = True
+        return driver
+
+    def _log_in(self):
+        """
+        logs into "the Hiking Project"'s website
+        """
+        print("-- log in")
+        self._homepage()
+        elem = self._driver.find_element_by_link_text("Sign In")
+        elem.click()
+        elem_email = self._driver.find_element_by_xpath("//input[@placeholder='Log in with email']")
+        elem_email.clear()
+        elem_email.send_keys("matan.pinkas@mail.huji.ac.il")
+        elem_pw = self._driver.find_element_by_xpath("//input[@placeholder='Password']")
+        elem_pw.clear()
+        elem_pw.send_keys("313287237mM")
+        elem = self._driver.find_elements_by_xpath("//button[@class='btn btn-primary btn-lg']")
+        for e in elem:
+            if e.text == "Log In":
+                elem = e
+        elem.click()
+
+    def _homepage(self):
+        """
+        navigates to "the hiking project"'s homepage
+        """
+        self._driver.get("https://www.hikingproject.com/")
+
+# collects data #
+    def _trails_in_urls(self):
+        """
+        return a list of urls (strings) for tracks in the current country
+        """
+        print("-- collecting urls")
+
+        self._homepage()
+        query = "//a[@title='" + self._country + "']"
+        elem = self._driver.find_element_by_xpath(query)
+        url = elem.get_attribute("href")
+        self._driver.get(url)
+        # driver is not at page with data on paths in <name>
+        # loop makes page show all trails in <name>
+
+        while True:
+            try:
+                show_more = self._driver.find_element_by_xpath("//button[@id='load-more-trails']")
+                show_more.click()
+            except selenium.common.exceptions.NoSuchElementException:
+                break
+        trail_elements = self._driver.find_elements_by_xpath("//tr[@class='trail-row']")
+        trail_urls = []
+        for i in range(len(trail_elements)):
+            trail_urls.append(trail_elements[i].get_attribute("data-href"))
+        return trail_urls
+
+    def _get_page_data(self):
+        """
+        gets data from a trail page in the hiking project: track's length and track's difficulty.
+        :returns track_len: strings that holds a double
+                track_dif: string representation of enum of class TrackDifficulty
+        """
+        e = self._driver.find_elements_by_xpath("//div[@class='stat-block mx-1 pb-2']")[0]
+        txt = e.text
+        txt = txt.split()
+        track_len = txt[1]
+        track_dif = txt[-1]
```

Comment:
I don't like this variable name dif could be difficulty or difference which makes it a bit confusing (even though there is documentation so I'm not sure it requires a name change (although I would at the very least change the name to track_diff just because it feels more natural)